jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_attributes.cpp

```diff
@@ -596,4 +597,32 @@ SCENARIO(
       }
     }
   }
+
+  GIVEN("A method that may or may not throw exceptions")
+  {
+    const symbol_tablet &new_symbol_table = load_java_class(
+      "ThrowsExceptions", "./java_bytecode/java_bytecode_parser");
+    WHEN("Parsing the exceptions attribute for a method that throws exceptions")
+    {
+      THEN("We should be able to get the list of exceptions it throws")
+      {
+        const symbolt &class_symbol =
+          new_symbol_table.lookup_ref("java::ThrowsExceptions");
+        const symbolt &method_symbol =
+          new_symbol_table.lookup_ref("java::ThrowsExceptions.test:()V");
+        const code_typet method = to_code_type(method_symbol.type);
+        const std::vector<irept> exceptions = method.throws_exceptions();
+        REQUIRE(exceptions.size() == 2);
```

Comment:
Nit: you can use `Catch:Matchers::Vector` to do this in a nicer way - when I have an IDE I can show you a snippet, else you can search the code base for use cases. Not blocking but makes unit test and errors from it nicer to read.